From 6fb16b64494733fe5b8b037675cab82a0cc59fa0 Mon Sep 17 00:00:00 2001
From: Pierrick Bouvier <pierrick.bouvier@linaro.org>
Date: Wed, 29 Oct 2025 06:27:29 -0700
Subject: [PATCH] lock interface: set report with correct regions for QEMU Sata
 device

Report only BAR[5], as BAR[4] are I/O ports

01:00.0 SATA controller: Intel Corporation 82801IR/IO/IH (ICH9R/DO/DH) 602) (prog-if 01 [AHCI 1.0])
    Region 4: I/O ports at 1000 [size=32]
    Region 5: Memory at 81000000 (32-bit, non-prefetchable) [size=4K]
---
 .../pci_tdisp_device_lock_interface.c         | 27 +++----------------
 1 file changed, 4 insertions(+), 23 deletions(-)

diff --git a/library/pci_tdisp_device_lib_sample/pci_tdisp_device_lock_interface.c b/library/pci_tdisp_device_lib_sample/pci_tdisp_device_lock_interface.c
index 9a4e1da..45f4466 100644
--- a/library/pci_tdisp_device_lib_sample/pci_tdisp_device_lock_interface.c
+++ b/library/pci_tdisp_device_lib_sample/pci_tdisp_device_lock_interface.c
@@ -11,7 +11,7 @@
 #include "library/spdm_transport_pcidoe_lib.h"
 #include "library/pci_tdisp_device_lib.h"
 
-#define MMIO_RANGE_COUNT 4
+#define MMIO_RANGE_COUNT 1
 #define DEVICE_INFO_LEN  16
 
 typedef struct {
@@ -85,29 +85,10 @@ libtdisp_error_code_t pci_tdisp_device_lock_interface (const void *pci_doe_conte
     interface_report->lnr_control = 0;
     interface_report->tph_control = 0;
     interface_report->mmio_range_count = MMIO_RANGE_COUNT;
-    interface_report->mmio_range[0].first_page = 0x0000;
+    interface_report->mmio_range[0].first_page = 0x81000000 >> 12;
     interface_report->mmio_range[0].number_of_pages = 1;
-    interface_report->mmio_range[0].range_attributes =
-        PCI_TDISP_MMIO_RANGE_ATTRIBUTES_IS_NON_TEE_MEM;
-    interface_report->mmio_range[0].range_id = 1;
-
-    interface_report->mmio_range[1].first_page = 0x8000;
-    interface_report->mmio_range[1].number_of_pages = 4;
-    interface_report->mmio_range[1].range_attributes =
-        PCI_TDISP_MMIO_RANGE_ATTRIBUTES_IS_MEM_ATTR_UPDATABLE;
-    interface_report->mmio_range[1].range_id = 2;
-
-    interface_report->mmio_range[2].first_page = 0x10000;
-    interface_report->mmio_range[2].number_of_pages = 8;
-    interface_report->mmio_range[2].range_attributes =
-        PCI_TDISP_MMIO_RANGE_ATTRIBUTES_IS_MEM_ATTR_UPDATABLE;
-    interface_report->mmio_range[2].range_id = 3;
-
-    interface_report->mmio_range[3].first_page = 0x20000;
-    interface_report->mmio_range[3].number_of_pages = 8;
-    interface_report->mmio_range[3].range_attributes =
-        PCI_TDISP_MMIO_RANGE_ATTRIBUTES_IS_MEM_ATTR_UPDATABLE;
-    interface_report->mmio_range[3].range_id = 4;
+    interface_report->mmio_range[0].range_attributes = 0;
+    interface_report->mmio_range[0].range_id = 5;
 
     interface_report->device_specific_info_len = DEVICE_INFO_LEN;
     libspdm_copy_mem (interface_report->device_specific_info,
-- 
2.47.3

